dpe/src/dpe_instance.rs

```diff
@@ -275,13 +284,41 @@ impl<C: Crypto> DpeInstance<'_, C> {
                 .update(&tci_bytes[..len])
                 .map_err(|_| DpeErrorCode::InternalError)?;
 
-            // Hash internal DPE info
-            if context.uses_internal_dpe_info {
-                let mut internal_dpe_info = [0u8; INTERNAL_DPE_INFO_SIZE];
-                self.serialize_internal_dpe_info(&mut internal_dpe_info);
-                hasher
-                    .update(&internal_dpe_info[..INTERNAL_DPE_INFO_SIZE])
-                    .map_err(|_| DpeErrorCode::InternalError)?;
+            // Check if any context uses internal inputs
+            uses_internal_input_info = uses_internal_input_info || context.uses_internal_input_info;
+            uses_internal_input_dice = uses_internal_input_dice || context.uses_internal_input_dice;
+        }
+
+        // Add internal input info to hash
+        if uses_internal_input_info {
+            let mut internal_input_info = [0u8; INTERNAL_INPUT_INFO_SIZE];
+            self.serialize_internal_input_info(&mut internal_input_info);
+            hasher
+                .update(&internal_input_info[..INTERNAL_INPUT_INFO_SIZE])
+                .map_err(|_| DpeErrorCode::InternalError)?;
+        }
+
+        // Add internal input dice to hash
+        if uses_internal_input_dice {
+            let ignored_locality = 0;
+            let mut offset = 0;
+            loop {
+                let cmd = GetCertificateChainCmd {
+                    size: MAX_CERT_SIZE.try_into().unwrap(),
+                    offset,
+                };
+                match cmd.execute(self, ignored_locality) {
+                    Ok(Response::GetCertificateChain(GetCertificateChainResp {
```

Comment:
Can this call `P::get_certificate_chain` instead? That will avoid the command structure setup, calling cmd.execute, etc.